server/src/main/java/nl/inl/blacklab/server/BlackLabServer.java

```diff
@@ -104,6 +105,8 @@ private void readConfig() throws BlsException {
 
             // Open log database
             try {
+                logDatabase = new ConsoleLogDatabase();
```

Comment:
This seems a touch suspicious given that BL is already logging `/docuser:8bd8a52e-dc77-4149-a305-0aab8aadbee3_NIGC/hits`... Is the `if (sqliteDatabase != null) {` check always going to fail after this? If not, is it okay that `logDatabase` is overwritten? There's probably good reasons for all of this. Would be helpful if you could add some comments explaining what's up.

Reply:
I know a bit surpising.

Blacklab implements its own "structured" logging, which is handled via configurations and it is a completely different code path than your typical logger (for example the logs you see coming from requests `/docuser:`). This change enables us to get to some of those logs, which will come handy when we have to understand why certain rules run slower. The implementation of this logging is set to `trace` so they are disabled by default. 

The `if (sqliteDatabase != null) ` always fails since we don't configured this type of logging for our blacklab instances. Notice the statements prior to the if won't make the if statement fail, since they are condition on the config  not on the state of the `searchManager`

Reply:
Enabling  the db based tracing in production is a non-starter in my opinion. Even in development is a bit too much since 80%? of the information can probably bee seen just by looking at the logs.

Reply:
Thanks! So this is in some sense just a logging perf improvement unrelated to the rest of the code. Do I have that right?

It would be great to get some of this as a comment in the code.